frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
+                            filterIcon={filterIcon}
+                            rowSpan={2}
+                        />
+                        <th colSpan={3}>
+                            <p>File Location</p>
+                        </th>
+                        <MessageReceiversCol
+                            columnHeaderTitle={ColumnDataEnum.TransportResults}
+                            activeColumn={activeColumn}
+                            setActiveColumn={setActiveColumn}
+                            activeColumnSortOrder={activeColumnSortOrder}
+                            setActiveColumnSortOrder={setActiveColumnSortOrder}
+                            filterIcon={filterIcon}
+                            rowSpan={2}
+                        />
+                    </tr>
+                    <tr>
+                        <MessageReceiversCol
+                            columnHeaderTitle={ColumnDataEnum.Main}
+                            activeColumn={activeColumn}
+                            setActiveColumn={setActiveColumn}
+                            activeColumnSortOrder={activeColumnSortOrder}
+                            setActiveColumnSortOrder={setActiveColumnSortOrder}
+                            filterIcon={filterIcon}
+                            rowSpan={1}
+                        />
+                        <MessageReceiversCol
+                            columnHeaderTitle={ColumnDataEnum.Sub}
+                            activeColumn={activeColumn}
+                            setActiveColumn={setActiveColumn}
+                            activeColumnSortOrder={activeColumnSortOrder}
+                            setActiveColumnSortOrder={setActiveColumnSortOrder}
+                            filterIcon={filterIcon}
+                            rowSpan={1}
+                        />
+                        <MessageReceiversCol
+                            columnHeaderTitle={ColumnDataEnum.FileName}
+                            activeColumn={activeColumn}
+                            setActiveColumn={setActiveColumn}
+                            activeColumnSortOrder={activeColumnSortOrder}
+                            setActiveColumnSortOrder={setActiveColumnSortOrder}
+                            filterIcon={filterIcon}
+                            rowSpan={1}
                         />
+                    </tr>
+                </thead>
+                <tbody className="message-receivers-table">
+                    {sortedData.map(
+                        (receiver: NormalizedReceiverData, index) => (
+                            <MessageReceiversRow
+                                key={index}
+                                receiver={receiver}
+                                activeColumn={activeColumn}
+                                activeColumnSortOrder={activeColumnSortOrder}
+                                setModalText={setModalText}
```

Comment:
Rather than pass `setModalText` and `modalRef` down directly, can we just wrap that in a function?

```jsx
function openModal({ title, body }) {
    setModalText({ title, body });
    modalRef?.current?.toggleModal();
}
```

Then we could generalize this to an optional `onCellClick` for the individual cells wherever needed 🤷

Reply:
Added to our discussion on what features to add for a genericized table component: https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/gh/cdcgov/prime-reportstream/8280